modules/users/tests/server/user-block.server.routes.tests.js

```diff
@@ -0,0 +1,279 @@
+    // Alice user
+    const aliceCredentials = {
+      username: 'alice_the_blocker',
+      password: 'TR-I$Aw3$0m4',
+    };
+
+    const aliceProfile = {
+      public: true,
+      firstName: 'Alice',
+      lastName: 'Doe',
+      displayName: 'Alice Doe',
+      email: 'alice@example.org',
+      emailToken: 'initial email token',
+      username: aliceCredentials.username.toLowerCase(),
+      password: aliceCredentials.password,
+      provider: 'local',
+    };
+    const aliceUser = new User(aliceProfile);
+    alice = {
+      credentials: aliceCredentials,
+      user: aliceUser,
+      profile: aliceProfile,
+    };
+
+    // Bob user
+    const bobCredentials = {
+      username: 'bob_the_blocked',
+      password: 'TR-I$Aw3$0m4',
+    };
+    const bobProfile = {
+      public: true,
+      firstName: 'Bob',
+      lastName: 'Doe',
+      displayName: 'bob doe',
+      email: 'bob@example.org',
+      emailToken: 'initial email token',
+      username: bobCredentials.username.toLowerCase(),
+      password: bobCredentials.password,
+      provider: 'local',
+    };
+    const bobUser = new User(bobProfile);
+    bob = { credentials: bobCredentials, user: bobUser, profile: bobProfile };
+
+    // Carol user
+    const carolCredentials = {
+      username: 'carol_the_blocker',
+      password: 'TR-I$Aw3$0m4',
+    };
+    const carolProfile = {
+      public: true,
+      firstName: 'Carol',
+      lastName: 'Doe',
+      displayName: 'Carol Doe',
+      email: 'carol@example.org',
+      emailToken: 'initial email token',
+      username: carolCredentials.username.toLowerCase(),
+      password: carolCredentials.password,
+      provider: 'local',
+    };
+    const carolUser = new User(carolProfile);
+    carol = {
+      credentials: carolCredentials,
+      user: carolUser,
+      profile: carolProfile,
+    };
+
+    // Save alice and bob to test db
+    Promise.all([aliceUser.save(), bobUser.save(), carolUser.save()])
+      .then(() => done())
+      .catch(done);
+  });
+
+  it('should be able to see a user if not blocked by her', function (done) {
+    /* bob login */
+    agent
+      .post('/api/auth/signin')
+      .send(bob.credentials)
+      .expect(200)
+      .end(function (err) {
+        // Handle signin error
+        if (err) {
+          return done(err);
+        }
+        /* bob can see alice */
+        agent
+          .get('/api/users/' + alice.credentials.username)
+          .expect(200)
+          .end(function (err, resp) {
+            if (err) {
+              return done(err);
+            }
+            const response = resp.body;
+            should(response).have.property('displayName');
+            should(response.displayName).be.equal('Alice Doe');
+            return done();
+          });
+      });
+  });
+
+  it('should not see a user if users blocked by her', function (done) {
+    /* alice login */
+    agent
+      .post('/api/auth/signin')
+      .send(alice.credentials)
```

Comment:
Some examples how to simplify this nowadays:

https://github.com/Trustroots/trustroots/blob/aaa8b9ed0269c8e4ea157bcc6097db18b7eb5719/modules/admin/tests/server/admin.users.server.routes.tests.js#L115-L122

https://github.com/Trustroots/trustroots/blob/aaa8b9ed0269c8e4ea157bcc6097db18b7eb5719/modules/references/tests/server/reference-read-one.server.routes.tests.js#L78-L80

Not a blocker for merging. :-)